db/db_impl_files.cc

```diff
@@ -456,7 +456,9 @@ void DBImpl::PurgeObsoleteFiles(JobContext& state, bool schedule_only) {
     } else {
       dir_to_sync =
           (type == kLogFile) ? immutable_db_options_.wal_dir : dbname_;
-      fname = dir_to_sync + "/" + to_delete;
+      fname = dir_to_sync
+            + (dir_to_sync.back() == '/' || to_delete.front() == '/' ? "" : "/")
+            + to_delete;
```

Comment:
assert dir_to_sync and to_delete are not empty.
Actually, maybe dir_to_sync can be empty?

Reply:
This is my last comment. If you address this, I'll defer to @riversand963 to accept and merge.

Reply:
A new commit is pushed.

Actually, I think maybe we need something like `canonicalize_file_name()` or `realpath()`.

Thanks for your review.